### PR TITLE
Process cells

### PR DESCRIPTION
**Pull request by jordansread**
This has code from #26 , but I needed something in fetch to get it working, so that is why it is there. 

This just builds the cells for getting precip.

### CODE REVIEW

**Comment by jordansread** [comment]
image

File: scripts/process/grid_from_mask.R [hunk 1]
```diff
@@ -0,0 +1,21 @@
+#' take an sf object polygon and make a grid out of it
```
Comment (dblodgett-usgs) [inline scripts/process/grid_from_mask.R:1]: Could use a bit more for the uninitiated.

File: scripts/process/grid_from_mask.R [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  deps <- readDepends(viz)
+  checkRequired(viz, c("cell_size"))
+  
+  clip_poly <- deps[['clip_poly']]
+  crs <- sf::st_crs(clip_poly)
+  cell_size <- viz[["cell_size"]]
+  
+  # this is a simple way to dissolve multiple polygons into a single mask
+  clip_mask <- clip_poly %>% 
+    mutate(group=1) %>% group_by(group) %>% summarise()
```
Comment (dblodgett-usgs) [inline scripts/process/grid_from_mask.R]: Isn't this what sf::st_union() does?  
From the docs:  
> If st_union is called with a single argument, x, (with y missing) and by_feature is FALSE all geometries are unioned together and an sfg or single-geometry sfc object is returned.

File: scripts/process/grid_from_mask.R [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  cell_grid <- sf::st_make_grid(clip_mask, cellsize = cell_size, crs = crs)
+  cell_grid_clipped <- sf::st_intersection(cell_grid, clip_mask)
```
Comment (dblodgett-usgs) [inline scripts/process/grid_from_mask.R]: In the past we've also wanted the cell centroids... Not sure if there is resistance to creating a list containing two or three "versions" of these cells as output, but the unclipped version, the clipped version, and the cell centers for all cells that are partially in the clipped version might be handy?

**Review by dblodgett-usgs — APPROVED** [review]
I think this looks good. Probably a bit more to do to be complete, but the basic pattern is solid. Take a look at sf::st_union, I think it'll do what you need for merging polygons.

File: scripts/process/grid_from_mask.R [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  cell_size <- viz[["cell_size"]]
+  
+  # this is a simple way to dissolve multiple polygons into a single mask
+  clip_mask <- clip_poly %>% st_buffer(dist = 0) %>% 
+    mutate(group='mask') %>% group_by(group) %>% summarise()
+  cell_grid <- sf::st_make_grid(clip_mask, cellsize = cell_size, crs = crs)
+  cell_grid_clipped <- sf::st_intersection(cell_grid, clip_mask)
+  
+  # add an id to be used for adding data later:
+  cell_grid_clipped$id <- paste('cell_', 1:length(cell_grid_clipped), sep= '')
```
Comment (jordansread) [inline scripts/process/grid_from_mask.R]: I guess this doesn't work and modifies the object in a weird way. plotting works if you skip this line, but if you include it:
```r
plot(cell_grid_clipped, col = 'red')
Error in UseMethod("st_bbox") : 
  no applicable method for 'st_bbox' applied to an object of class "character"
```
and while dplyr verbs work for some `sf` classes, not this one:
```r
cell_grid_clipped <- mutate(cell_grid_clipped, id = 'test')
Error in UseMethod("mutate_") : 
  no applicable method for 'mutate_' applied to an object of class "c('sfc_GEOMETRY', 'sfc')"
```

**Pull request merged**